src/main/java/decodes/cwms/CwmsSiteDAO.java

```diff
@@ -488,29 +494,38 @@ public void fillCache()
 				{
 					warning("SiteName for id=" + key + " (" + nameType + ":"
 						+ nameValue + ") but no matching site.");
-					continue;
+					return;
```

Comment:
The 22 line lambda might be simplified, and potential boost (for large list of sites) if the siteList were a dictionary/hash (might still need List for other uses)

Reply:
I had a similar idea. But first I want to see if I can just make that query a full on join of thing as it would let us go through the loop in one shot.

Reply:
the trick with the join is that JDBC doesn't keep the table prefix name so you have to manually add it. Bit arduous so I'm likely to wait until/if JDBI gets used as it has features to handle that particular issue.